docs/csharp/programming-guide/concepts/async/async-return-types.md

```diff
@@ -23,313 +23,75 @@ translation.priority.mt:
   - "tr-tr"
 ---
 # Async Return Types (C#)
-Async methods have three possible return types: <xref:System.Threading.Tasks.Task%601>, <xref:System.Threading.Tasks.Task>, and void. In Visual Basic, the void return type is written as a [Sub](../../../../visual-basic/programming-guide/language-features/procedures/sub-procedures.md) procedure. For more information about async methods, see [Asynchronous Programming with async and await (C#)](../../../../csharp/programming-guide/concepts/async/index.md).  
+Async methods can have the following return types:
+
+- <xref:System.Threading.Tasks.Task%601>, for an async method that returns a value. 
+ 
+-  <xref:System.Threading.Tasks.Task>, for an async method that performs an operation but returns no value.
+
+- `void`, for an event handler. 
+
+- Any type that has an accessible `GetAwaiter` method (starting with C# 7).
```

Comment:
I'd expand this to say that the object returned by the `GetAwaiter` method must implement the `ICriticalNotifyCompletion` interface.